packages/components/src/icon/with-currentcolor.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { cloneElement } from '@wordpress/element';
+
+/**
+ * Modifies an element to use the `currentColor` for the fills in its SVG descendants.
+ *
+ * In most cases you should be able to use the `currentColor` prop of the `Icon` component instead.
+ *
+ * ```jsx
+ * import { wordpress } from '@wordpress/icons';
+ * import { RangeControl } from '@wordpress/components';
+ *
+ * <RangeControl beforeIcon={ withCurrentColor( wordpress ) } />
```

Comment:
This is a niche HOF designed to be an ergonomic way to feed `@wordpress/icons` arguments into `icon` props on components like RangeControl, where they pass that `icon` straight into an `<Icon>` but do not set any `currentColor` styles.

It should be rare to have to use this, because most components that pass an `icon` argument straight into an `<Icon>` already have `currentColor` styles set (e.g. Button).